### PR TITLE
Fix Improved Healthstone talent for Soulwells

### PR DESCRIPTION
**Pull request by Warlockbugs**
Summons improved Soulwells instead of regular one if caster has the talent.

This may be a bit hacky, but i couldn't find a prettier way to achieve the same result.

### CODE REVIEW

File: src/game/SpellEffects.cpp [hunk 1]
```diff
@@ -7220,6 +7220,14 @@ void Spell::EffectTransmitted(SpellEffectIndex eff_idx)
 {
     uint32 name_id = m_spellInfo->EffectMiscValue[eff_idx];
 
+    // Improved Healthstone Talent Support for Soulwell
+    if (m_caster && name_id == 181621) {
```
Comment (robinsch) [inline src/game/SpellEffects.cpp]: Codestyle {

**Comment by Warlockbugs** [comment]
Aye, an old habit for Qt style blocks. Fixed.

**Comment by xfurry** [comment]
Hmm... quite hack.
Let's see if we can find a way around this and use only spell ids, preferably.

**Comment by robinsch** [comment]
I'm not really sure, I mean when player logs off you can no longer get improved Stones.

**Comment by robinsch** [comment]
@xfurry What other way? The spell should summon a diffrent gameobject when having a certain aura (talent). I have no idea how to handle this otherwise.

**Comment by killerwife** [comment]
the only other way is to edit SpellInfo during runtime for specific player, so that the GO summoned in effect is the one of choice, but since we store spellinfo globally thats just not possible
it is very likely we "overload" stuff in a wrong way, kinda made apparent with the whole "overload class scripts" effects,

**Comment by robinsch** [comment]
Which will basicly result in the same check?

**Comment by killerwife** [comment]
Unless you propose a runtime solution, which on apply of talent aura, you would change a value in the memory for given player which would edit GO summoned for this particular spell, you will end up with some flavor of the same check anyway. I will try to pull an idea out of my arse in the future for this, but for now, this PR is legit.

**Comment by xfurry** [comment]
There is no "normal" way. But something like this looks better.
Instead of checking for the GO entry, we check for the spell entry. Makes more sense.

```diff
@@ -11416,10 +11416,23 @@ void Spell::EffectModifyThreatPercent(SpellEffectIndex /*eff_idx*/)
 
 void Spell::EffectTransmitted(SpellEffectIndex eff_idx)
 {
     uint32 name_id = m_spellInfo->EffectMiscValue[eff_idx];
 
+    switch (m_spellInfo->Id)
+    {
+        case 29886:
+            if (m_caster->HasAura(18692))
+                name_id = 183510;
+            else if (m_caster->HasAura(18693))
+                name_id = 183511;
+            break;
+        case 58889:
+            // ToDo: something with GOs 193170 and 193171
+            break;
+    }
+
     GameObjectInfo const* goinfo = ObjectMgr::GetGameObjectInfo(name_id);
 
     if (!goinfo)
     {
         sLog.outErrorDb("Gameobject (Entry: %u) not exist and not created at spell (ID: %u) cast", name_id, m_spellInfo->Id);

```

**Comment by Warlockbugs** [comment]
Updated to switch.

**Comment by xfurry** [comment]
Poke @killerwife about the ToDo: note from my patch. Any idea how would that work?

**Comment by robinsch** [comment]
You could also move this to gameobject scripts.

**Comment by killerwife** [comment]
@xfurry same rules for that TODO, the GOs are ordered by numbers for the talents

Also, this will be easily made more nice once we add my spell_scripting implementation. It also gave me an idea how to generically pass parameters from my overloading ^^

**Pull request closed without merging**